### PR TITLE
Improving Array module documentation

### PR DESCRIPTION
**Pull request by fmpanelli**
Adding descriptions and examples for several functions of the Array module.
My aim is to make the awesome `fp-ts` library a bit more accessible to users and to beginners in particular.

No changes have been done to the actual code.

Thanks in advance for every feedback and input for additional improvement.

Filippo

### CODE REVIEW

**Review by grossbart — COMMENTED** [review]
Thanks, this is very helpful, I think! Please have a look at my comments and fix them, if you agree. After that I would suggest to merge this.

File: src/Array.ts [hunk 1]
```diff
@@ -1,4 +1,7 @@
 /**
+ * The `Array` module enables dealing with Typescript's `Array<T>`
+ * type in a functional way, basically treating it as a `Functor` in `T`.
+ *
```
[the remark this answers is not in the document]
Comment (fmpanelli) [reply src/Array.ts:4]: I'll try and include both the beginner-friendly version and the fp-friendly versions 😀

File: src/Array.ts [hunk 2]
```diff
@@ -148,13 +172,31 @@ export const makeBy = <A>(n: number, f: (i: number) => A): Array<A> => (n <= 0 ?
...
+ * @example
+ * import { fromPredicate } from 'fp-ts/Array'
+ * import { pipe } from 'fp-ts/function'
+ *
+ * const isString = (x: string | number): x is string => typeof x === "string";
+ * assert.deepStrictEqual(pipe("a", fromPredicate(isString)), ["a"]);
+ * assert.deepStrictEqual(pipe(7, fromPredicate(isString)), []);
+ *
+ * const isPositiveNumber = (x: string | number) => typeof x === "number" && x>0;
```
[the remark this answers is not in the document]
Comment (fmpanelli) [reply src/Array.ts:195]: The idea was to show  a `Refinement` and a `Predicate`

**Comment by fmpanelli** [comment]
Included all improvements suggested by @grossbart , thank you!

**Pull request merged**